### PR TITLE
Feedbacks

### PR DESCRIPTION
**Pull request by SvetilnikST**

### CODE REVIEW

File: src/main/java/innowise/zuevsky/helpdesk/service/FeedbackService.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                new FeedbackNotFoundException("Feedback Not Found!")));
+    }
+
+    public FeedbackDto getFeedbackByTicketId(Long ticketId){
+        return  feedbackMapper.mapFeedbackToFeedbackDto( feedbackRepository.findFeedbackByTicketId(ticketId).orElseThrow(()->
+                        new FeedbackNotFoundException("Not found feedback with this ticketId")));
+    }
+
+    public void saveFeedback(FeedbackSaveDto saveFeedbackDto){
+        Optional<TicketDto> ticket = Optional.ofNullable(ticketsService.getTicket(saveFeedbackDto.getTicketId()));
```
Comment (kislyakdmitry) [inline src/main/java/innowise/zuevsky/helpdesk/service/FeedbackService.java:37]: Optional убрать

File: src/main/java/innowise/zuevsky/helpdesk/service/FeedbackService.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        ticket.ifPresent(ticketDto -> {
+            if(!ticketDto.getState().equals(State.DONE))  throw new FeedbackServiceException("Status for ticket is not DONE.");
```
Comment (kislyakdmitry) [inline src/main/java/innowise/zuevsky/helpdesk/service/FeedbackService.java:39]: if (condition) {
}

Comment (kislyakdmitry) [inline src/main/java/innowise/zuevsky/helpdesk/service/FeedbackService.java:39]: if (!) -> if()

**Review by kislyakdmitry — CHANGES_REQUESTED** [review]

File: src/main/java/innowise/zuevsky/helpdesk/controller/AuthenticationRestController.java [hunk 3]
```diff
@@ -35,7 +36,7 @@
         try {
             authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(request.getEmail(), request.getPassword()));
             User user = usersRepository.findByEmail(request.getEmail()).orElseThrow(() ->
-                    new UsernameNotFoundException("User doesn't exists"));
+                    new UserNotFoundException("User doesn't exists"));
```
[the remark this answers is not in the document]
Comment (SvetilnikST) [reply src/main/java/innowise/zuevsky/helpdesk/controller/AuthenticationRestController.java:39]: not for me.

File: src/main/java/innowise/zuevsky/helpdesk/service/FeedbackService.java (hunk 2 above)
Comment (SvetilnikST) [reply src/main/java/innowise/zuevsky/helpdesk/service/FeedbackService.java]: Fixed.

**Pull request merged**